tests/queries/0_stateless/03277_empty_tuple_formats.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+
+# Test that no formats crash or return LOGICAL_ERROR on empty tuple.
+# Some formats read zero rows; this test currently allows that.
+
+FILE=03277_$CLICKHOUSE_DATABASE
+
+# With schema inference.
+for format in Native TSV CSV TSKV JSON JSONCompact JSONEachRow JSONObjectEachRow JSONCompactEachRow JSONColumns JSONCompactColumns JSONColumnsWithMetadata ORC Arrow
+do
+  echo $format
+  $CLICKHOUSE_LOCAL -q "
+    insert into function file('$FILE', '$format') select () settings engine_file_truncate_on_insert=1;
+    select * from file('$FILE', '$format');"
+done
+
+# Picky about column names.
+echo Avro
+$CLICKHOUSE_LOCAL -q "
+  insert into function file('$FILE', 'Avro') select () as x settings engine_file_truncate_on_insert=1;
+  select * from file('$FILE', 'Avro');"
+
+# Without schema inference.
+for format in RowBinary Values BSONEachRow MsgPack Native TSV CSV TSKV JSON JSONCompact JSONEachRow JSONObjectEachRow JSONCompactEachRow JSONColumns JSONCompactColumns JSONColumnsWithMetadata ORC Arrow
+do
+  echo $format
+  $CLICKHOUSE_LOCAL -q "
+    insert into function file('$FILE', '$format', 'x Tuple()') select () as x settings engine_file_truncate_on_insert=1;
+    select * from file('$FILE', '$format', 'x Tuple()');"
+done
+
+# Formats that don't support empty tuples.
+$CLICKHOUSE_LOCAL -q "
+  insert into function file('$FILE', 'Parquet') select () settings engine_file_truncate_on_insert=1; -- {serverError BAD_ARGUMENTS}
```

Comment:
How does it come that this results in `BAD_ARGUMENTS`, while the similar query in `02735_parquet_encoder.sql` returns `PARQUET_EXCEPTION`?

Reply:
Oh, `02735_parquet_encoder.sql` failed:
```
024-11-29 03:46:31 Expected server error code: 723 but got: 36 (query: insert into function file(empty_tuple_02735.parquet) select (); -- {serverError PARQUET_EXCEPTION}).
2024-11-29 03:46:31 Received exception from server (version 24.12.1):
2024-11-29 03:46:31 Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Parquet doesn't support empty tuples. (BAD_ARGUMENTS)
2024-11-29 03:46:31 (query: insert into function file(empty_tuple_02735.parquet) select (); -- {serverError PARQUET_EXCEPTION})
```
So I guess that should be changed.